src/PythonProjectBootstrapper/ProjectGenerationUtils.py

```diff
@@ -0,0 +1,204 @@
+# ----------------------------------------------------------------------
+# |
+# |  Copyright (c) 2024 Scientific Software Engineering Center at Georgia Tech
+# |  Distributed under the MIT License.
+# |
+# ----------------------------------------------------------------------
+import hashlib
+import itertools
+import os
+import sys
+from pathlib import Path
+
+from rich import print  # pylint: disable=redefined-builtin
+from rich.panel import Panel
+import yaml
+
+from dbrownell_Common import PathEx
+from PythonProjectBootstrapper import __version__
+
+# The following imports are used in cookiecutter hooks. Import them here to
+# ensure that they are frozen when creating binaries,
+import shutil  # pylint: disable=unused-import, wrong-import-order
+import textwrap  # pylint: disable=unused-import, wrong-import-order
+
+
+HASH_ALG = hashlib.sha256
```

Comment:
In the current implementation, the caller has to provide this value when calling `_GenerateFileHash`. Instead, default the `hash_fn` parameter in `_GenerateFileHash` to `hashlib.sha256` and remove this value.

Reply:
Done